lib/Platform/Intl/java/com/facebook/hermes/intl/UnicodeExtensionKeys.java

```diff
@@ -0,0 +1,177 @@
+package com.facebook.hermes.intl;
+
+import android.icu.util.ULocale;
+import android.os.Build;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class UnicodeExtensionKeys {
+    public static String CALENDAR = "calendar";
+    public static String CALENDAR_CANON = "ca";
+
+    public static String NUMERINGSYSTEM = "numbers";
+    public static String NUMERINGSYSTEM_CANON = "nu";
+
+    public static String HOURCYCLE = "hours";
+    public static String HOURCYCLE_CANON = "hc";
+
+    public static String COLLATION = "collation";
+    public static String COLLATION_CANON = "co";
+
+    public static String COLLATION_NUMERIC = "colnumeric";
+    public static String COLLATION_NUMERIC_CANON = "kn";
+
+    public static String COLLATION_CASEFIRST = "colcasefirst"; // TODO:: double check this
+    public static String COLLATION_CASEFIRST_CANON = "kf";
+
+    // TODO :: Build/Look-for a BiMap datastructure.
+    private static HashMap<String, String> s_canonicalkey_icukey_map;
+    private static HashMap<String, String> s_icukey_canonicalkey_map;
+
+    static {
+        s_canonicalkey_icukey_map = new HashMap<String, String>() {{
```

Comment:
TODO:: These maps should be part of the Intl service instance .. not global .. so that their cost is per instance .. and garbage collectable ..